cube/src/main/java/org/jboss/arquillian/ce/cube/CEEnvironmentProcessor.java

```diff
@@ -0,0 +1,281 @@
+ * StartCube/StopCube).
+ * 
+ * @author Rob Cernich
+ */
+public class CEEnvironmentProcessor {
+
+    private final Logger log = Logger.getLogger(CEEnvironmentProcessor.class.getName());
+    
+    public interface TemplateDetails {
+        public Map<String,String> getLabels();
+        public int getReplicas();
+    }
+
+    @Inject @ClassScoped
+    private InstanceProducer<TemplateDetails> templateDetailsProducer;
+
+    /**
+     * Create the environment as specified by @Template or
+     * arq.extension.ce-cube.openshift.template.* properties.
+     * 
+     * In the future, this might be handled by starting application Cube
+     * objects, e.g. CreateCube(application), StartCube(application)
+     * 
+     * Needs to fire before the containers are started.
+     */
+    public void createEnvironment(@Observes(precedence=10) BeforeClass event, OpenShiftAdapter client,
+            CECubeConfiguration configuration) throws DeploymentException {
+        final TestClass testClass = event.getTestClass();
+        log.info(String.format("Creating environment for %s", testClass.getName()));
+        OpenShiftResourceFactory.createResources(testClass.getName(), client, null, testClass.getJavaClass(),
+                configuration.getProperties());
+        processTemplate(testClass, client, configuration);
+    }
+
+    /**
+     * Wait for the template resources to come up after the test container has
+     * been started. This allows the test container and the template resources
+     * to come up in parallel.
+     */
+    public void waitForDeployments(@Observes(precedence = -100) BeforeDeploy event, OpenShiftAdapter client,
+            TemplateDetails details, TestClass testClass) throws Exception {
+        if (details == null) {
+            return;
+        }
+        log.info(String.format("Waiting for environment for %s", testClass.getName()));
+        try {
+            delay(client, details.getLabels(), details.getReplicas());
+        } catch (Throwable t) {
+            throw new DeploymentException("Error waiting for template resources to deploy: " + testClass.getName(), t);
+        }
+    }
+
+    /**
+     * Tear down the environment.
+     * 
+     * In the future, this might be handled by stopping application Cube
+     * objects, e.g. StopCube(application), DestroyCube(application).
+     */
+    public void deleteEnvironment(@Observes AfterClass event, OpenShiftAdapter client) throws Exception {
+        final TestClass testClass = event.getTestClass();
+        client.deleteTemplate(testClass.getName());
+        OpenShiftResourceFactory.deleteResources(testClass.getName(), client);
+    }
+
+    private void processTemplate(TestClass tc, OpenShiftAdapter client, CECubeConfiguration configuration)
+            throws DeploymentException {
+        final StringResolver resolver = Strings.createStringResolver(configuration.getProperties());
+        final Template template = tc.getAnnotation(Template.class);
```

Comment:
Yeah, some multi-test-suite would be nice.

I have something similar here in GAE TCK
* https://github.com/GoogleCloudPlatform/appengine-tck/tree/master/common/appengine-tck-multisuite/src/main/java/com/google/appengine/tck/multisuite

If this could be of use ...